homeassistant/components/ps4/__init__.py

```diff
@@ -31,3 +33,40 @@
     await hass.config_entries.async_forward_entry_unload(
         entry, 'media_player')
     return True
+
+
+async def async_migrate_entry(hass, entry):
+    """Migrate old entry."""
+    from pyps4_homeassistant.media_art import COUNTRIES
+
+    config_entries = hass.config_entries
+    data = entry.data
+    version = entry.version
+
+    reason = {1: "Region codes have changed"}
+
+    # Migrate Version 1 -> Version 2
+    if version == 1:
+        loc = location.detect_location_info()
```

Comment:
This does I/O so must be scheduled on the executor thread pool with `hass.async_add_executor_job`.